packages/components/src/overlay/src/useOverlayPosition.ts

```diff
@@ -45,15 +45,15 @@ export function useOverlayPosition({
             enabled: allowFlip,
             name: "flip",
             options: {
-                boundary: boundaryElement
+                boundary: boundaryElement ?? (isBrowser ? document.body : undefined)
```

Comment:
Should you add an utility function `getBodyElement` ? Maybe in the same file as `isBrowser`